README.md

```diff
@@ -202,7 +280,49 @@ algoliaResult =
     |> required "zip" string
 ```
 
-This will also make it easier to add [optional fields](http://package.elm-lang.org/packages/NoRedInk/elm-decode-pipeline/1.0.0/Json-Decode-Pipeline#optional) where necessary.
+This will also make it easier to add [optional fields](http://package.elm-lang.org/packages/NoRedInk/elm-decode-pipeline/latest/Json-Decode-Pipeline#optional) where necessary.
 
 [json2elm](http://json2elm.org/) can generate pipeline-style decoders from
 raw JSON.
+
+## Syntax
+
+### Use `case..of` over `if` where possible
+
+`case..of` is clever as it will generate more efficent JS, and it also allows you to catch unmatched patterns at compile time. It's also cheap to extend this data with something more useful later on, like if you need to add another branch. This saves code diffs.
+
+
+## Code Smells
+
+### If a module has a looong list of imports, consider refactoring
+
+Having complicated imports hurts our compile time! I don't know what to say about this other than if you feel that there's something wrong with the top 40 lines of your module because of imports, then it might be time to move things out into another module. Trust your gut.
+
+### If a function can be pulled outside of a let binding, then do it
+
+Giant let bindings hurt readability and performance. The less nested a function, the less functions are used in generated code.
+
+The update function is especially prone to get longer and longer: keep it as small as possible. Smaller functions that don't live in a let binding are more reusable.
+
+### If your application has too many constructors for your Msg type, consider refactoring
+
+Large `case..of` statements hurts compile time. It might be possible that some of your constructors can be combined, for example `type Msg = Open | Close` could actually be `type Msg = SetOpenState Bool`
+
+
+## Tooling
+
+### Use [`elm-init-scripts`](https://github.com/NoRedInk/elm-init-scripts) to start your projects
```

Comment:
Hm, keeping the reference for now since we have a PR that graciously updates it for 0.18.